src/array/cuda/cusparse_dispatcher.cuh

```diff
@@ -34,6 +34,32 @@ struct CSRGEMM {
   }
 };
 
+#ifdef USE_FP16
+template <>
+struct CSRGEMM<__half> {
+  template <typename... Args>
+  static inline cusparseStatus_t bufferSizeExt(Args... args) {
+    // TODO(ndickson): There is no cusparseHcsrgemm2_bufferSizeExt, so a different
+    // implementation would be required.
+    BUG_IF_FAIL(false) << "CSRGEMM::bufferSizeExt does not support dtype half (FP16).";
```

Comment:
While functionally it doesn't matter, this will instruct users to file a bug report. Since this is expected behavior, we should use `LOG(FATAL)` instead:
```
LOG(FATAL) << "CSRGEMM::bufferSizeExt does not support dtype half (FP16).";
```